pkg/plugins/secret/dev/dev_test.go

```diff
@@ -0,0 +1,130 @@
+	os.Setenv("LOCAL_SEC_DIR", "./.nitric/")
+
+	AfterSuite(func() {
+		// Cleanup default secrect directory
+		os.RemoveAll("./.nitric/")
+	})
+
+	testSecret := secret.Secret{
+		Name: "Test",
+	}
+	testSecretVal := []byte("Super Secret Message")
+	When("Put", func() {
+		When("Putting a secret to a non-existent secret", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should successfully store a secret", func() {
+				response, err := secretPlugin.Put(&testSecret, testSecretVal)
+				By("Not returning an error")
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("Returning a non nil response")
+				Expect(response).ShouldNot(BeNil())
+			})
+		})
+		When("Putting a secret to an existing secret", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should succesfully store a secret", func() {
+				response, err := secretPlugin.Put(&testSecret, testSecretVal)
+				By("Not returning an error")
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("Returning a non nil response")
+				Expect(response).ShouldNot(BeNil())
+			})
+		})
+		When("Putting a secret with an empty name", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should throw an error", func() {
+				emptySecretName := &secret.Secret{}
+				response, err := secretPlugin.Put(emptySecretName, testSecretVal)
+				By("Returning an error")
+				Expect(err).Should(HaveOccurred())
+
+				By("Returning a nil response")
+				Expect(response).Should(BeNil())
+			})
+		})
+	})
+	When("Get", func() {
+		When("Getting a secret that exists", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should return the secret", func() {
+				putResponse, _ := secretPlugin.Put(&testSecret, testSecretVal)
+				response, err := secretPlugin.Access(putResponse.SecretVersion)
+				By("Not returning an error")
+				Expect(err).ShouldNot(HaveOccurred())
+				By("Returning a response")
+				Expect(response.SecretVersion.Secret.Name).Should(Equal(testSecret.Name))
+				Expect(response.Value).Should(Equal(testSecretVal))
+			})
+		})
+		When("Getting a secret that doesn't exist", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should return an error", func() {
+				response, err := secretPlugin.Access(&secret.SecretVersion{
+					Secret: &secret.Secret{
+						Name: "test-id",
+					},
+					Version: "test-version-id",
+				})
+				By("Returning an error")
+				Expect(err).Should(HaveOccurred())
+				By("Returning a nil response")
+				Expect(response).Should(BeNil())
+			})
+		})
+		When("Getting a secret with an empty id", func() {
+			secretPlugin, _ := secretPlugin.New()
+			It("Should return an error", func() {
+				response, err := secretPlugin.Access(&secret.SecretVersion{
+					Secret:  &secret.Secret{},
+					Version: "test-version-id",
+				})
+				By("Returning an error")
+				Expect(err).Should(HaveOccurred())
+				By("Returning a nil response")
+				Expect(response).Should(BeNil())
+			})
+		})
+		When("Getting a secret with an empty version id", func() {
```

Comment:
Need a method and test for returning the 'latest' secret version

Reply:
Is this not already at line 87?